### PR TITLE
[ci] Add jobs for python 3.12-beta following its release

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :sparkles: New feature |

## Description

Required for https://github.com/pylint-dev/pylint/pull/8718. There's new construct in 3.12, the [type parameter syntax](https://docs.python.org/3.12/whatsnew/3.12.html#pep-695-type-parameter-syntax), to be added later.

### CODE REVIEW

**Review by DanielNoord — DISMISSED** [review]

**Comment by Pierre-Sassoulas** [comment]
Blocked by https://github.com/pytest-dev/pytest/pull/10894

**Comment by jacobtylerwalls** [comment]
Let's just add this [workaround](https://github.com/pydantic/pydantic/pull/5849/commits/662a576c9a7f392f9a943d9901345685d406ff84) from pydantic and merge?

**Comment by Pierre-Sassoulas** [comment]
We're going to have to handle ``visit_typealias`` before merging, the node generated are not the same in 3.12 even if we're not using the new syntax yet.

**Pull request closed without merging**
